packages/block/README.md

```diff
@@ -0,0 +1,48 @@
+> **Warning**:
+> This is a work in progress. The API is not stable and may change at any time.
+
+# `million/block`
+
+A low-level primitive for building compiled virtual DOM-based interactions.
+
+## Benchmarks
+
+<img src="https://cloud-nwwgel7ih-hack-club-bot.vercel.app/0image.png" width="300" />
+
+## Example Usage
+
+Inside the `/dev` folder, put the following code in `index.js`:
+
+```js
+import { createBlock, fragment } from 'packages/block';
+
+const block = createBlock((props) => {
+  // Any prop within props isn't the actual value, it's a "Hole"
+  // always interpolate into the JSX.
```

Comment:
Is that mean we are no longer able to pass the event handlers (functions) to the `block`?

Reply:
You can pass event handlers like:

```js
const block = createBlock(({ listener, value }) => <div onClick={listener} onMouseOver={() => alert('hi!')}>{value}</div>)
```

But this is not allowed:

```js
const block = createBlock(({ listener, value }) => <div onMouseOver={() => listener(params)}>{ value + 1 }</div>)
```

Reply:
We might have a hard time teaching people about this. A common mistake could be:

```
const block = createBlock(({ handleStateUpdate, value }) => {
  const handler = () => handleStateUpdate(value + 1);
  return <div onMouseOver={handler}>{ value + 1 }</div>;
})
```

Reply:
This is a good point, but the Block API is intended to be used as an implementation detail (either under runtime abstraction or compiler abstraction). 

One idea I have for this API is to create a `<Million>user code</Million>` component for React and Preact. This allows Million to have fast impact and easy integration whenever the developer requires faster rendering. Personally, this is a big advantage (compat libraries that Million, Inferno, and Preact have all have to try to maintain compat), but with this idea, it's just maintaining a React component.

The second is to build a compiler upon this API. A React component (or any component) can be compiled into blocks, allowing for very fast perf.